### PR TITLE
fix: correct CKBRPC.createBatchRequest() types

### PR DESCRIPTION
**Pull request by twhy**
# Description

This PR aims to correct `CKBRPC.createBatchRequest()` types.
```ts
// Before
createBatchRequest: <N extends keyof Base, P extends (string | number | object)[], R = any[]>(params?: any) => any;

// After
createBatchRequest: <N extends keyof Base, P extends (string | number | object)[], R = any[]>(params?: [method: N, ...rest: P][]) => [method: N, ...rest: P][] & {
    add: (n: N, ...p: P) => [method: N, ...rest: P][];
    remove: (index: number) => [method: N, ...rest: P][];
    exec: () => Promise<R>;
};
```

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?
```ts
$ npx pnpm build
$ npx pnpm test
```

### CODE REVIEW

File: .eslintrc.next.js [hunk 1]
```diff
@@ -23,6 +23,7 @@ module.exports = {
               1, // common for i + 1 in a loop
               2, // many .slice(2) since the '0x' prefix should be removed while calling 3rd-party library
               16, // toString(16)
+              30000, // rpc request timeout: 30 seconds
```
Comment (homura) [inline .eslintrc.next.js]: The number 30000 may not be common in the project. Should we name it directly?

```js
const DEFAULT_RPC_TIMEOUT = 30000;
```

**Review by homura — COMMENTED** [review]
Hi, can you add a changeset for `@ckb-lumos/rpc` for this PR?

```sh
npx changeset add
```

**Review by homura — COMMENTED** [review]
Hi, can you add a changeset for `@ckb-lumos/rpc` for this PR?

```sh
npx changeset add
```

**Comment by twhy** [comment]
> Hi, can you add a changeset for `@ckb-lumos/rpc` for this PR?
> 
> ```shell
> npx changeset add
> ```

Added.

File: .eslintrc.next.js (hunk 1 above)
Comment (twhy) [reply .eslintrc.next.js]: Done. Also updated `method.ts` to use `DEFAULT_RPC_TIMEOUT` .

**Review by homura — APPROVED** [review]

**Pull request merged**